lib/xandra/authenticator/password.ex

```diff
@@ -0,0 +1,26 @@
+defmodule Xandra.Authenticator.Password do
+  @moduledoc """
+  A `Xandra.Authenticator` that implements support for PasswordAuthenticator
+  to authenticate with Cassandra server.
+
+  ## Example
+
+    Xandra.start_link(
+      authentication: {
+        Xandra.Authenticator.Password,
+        username: "xandra",
+        password: "secret",
+      }
+    )
```

Comment:
This is my personal preference and I'd like @lexmag's input as well: I think the formatting is better as:

```elixir
authentication_options = [username: "xandra", password: "secret"]
Xandra.start_link(authentication: {Xandra.Authenticator.Password, authentication_options})
```

Thoughts?

(also small nitpick: we usually leave a blank line after a code sample and before `"""`)